src/waveform/waveformwidgetfactory.cpp

```diff
@@ -1153,14 +1166,23 @@ WaveformWidgetType::Type WaveformWidgetFactory::findTypeFromHandleIndex(int inde
 }
 
 int WaveformWidgetFactory::findHandleIndexFromType(WaveformWidgetType::Type type) {
-    int index = -1;
-    for (int i = 0; i < m_waveformWidgetHandles.size(); i++) {
-        WaveformWidgetAbstractHandle& handle = m_waveformWidgetHandles[i];
+    int index = 0;
+    for (const auto& handle : std::as_const(m_waveformWidgetHandles)) {
```

Comment:
if you're going to maintain an index like this, you might as well keep the original style of loop (and use .get() to get a const ref instead of mutable)

Reply:
Good point, I reverted this.
Though this is a QVector, not sure what you mean by `get()`.